sormas-ui/src/main/java/de/symeda/sormas/ui/utils/VaadinUiUtil.java

```diff
@@ -135,15 +138,42 @@ public static Window showPopupWindowWithWidth(Component content, String caption,
 	}
 
 	public static Window showModalPopupWindow(CommitDiscardWrapperComponent<?> content, String caption) {
-		return showModalPopupWindow(content, caption, null);
+		return showModalPopupWindow(content, caption, false);
+	}
+
+	public static Window showModalPopupWindow(CommitDiscardWrapperComponent<?> content, String caption, boolean discardOnClose) {
+		return showModalPopupWindow(content, caption, discardOnClose, null);
 	}
 
 	public static Window showModalPopupWindow(CommitDiscardWrapperComponent<?> content, String caption, Consumer<Boolean> callback) {
+		return showModalPopupWindow(content, caption, false, callback);
+	}
+
+	public static Window showModalPopupWindow(
+		CommitDiscardWrapperComponent<?> content,
+		String caption,
+		boolean discardOnClose,
+		Consumer<Boolean> callback) {
 		final Window popupWindow = VaadinUiUtil.showPopupWindow(content);
 		popupWindow.setCaption(caption);
 		content.setMargin(true);
 
-		content.addDoneListener(popupWindow::close);
+		final Registration closeListener;
+		if (discardOnClose) {
+			closeListener = popupWindow.addCloseListener((e) -> {
```

Comment:
Parantheses can be removed (e) -> e, and also curly braces are not necessary for one statement.